src/sessions/SessionManager.ts

```diff
@@ -218,16 +230,27 @@ export class SessionManager {
         }
     }
 
+    /**
+     * When the web client (and SessionManager) is initialized, a nil session is created.
+     * When the first event (usually the page view event) is recorded, a new session is created because the sessionId is nil.
+     * We ensure the nil session and new session have the same sampling decision if the nill session has not expired.
+     * All other instances where a new session is created will reevaluate the sampling decision to ensure that all sessions are randomly sampled.
+     */
     private createSession() {
+        this.sessionExpiry = new Date(
+            new Date().getTime() + this.config.sessionLengthSeconds * 1000
+        );
+
         this.session = {
             sessionId: v4(),
-            record: this.sample(),
+            record:
+                this.session.sessionId === NIL_UUID &&
+                new Date() < this.sessionExpiry
```

Comment:
issue: Isn't this always true, since we've updated `this.sessionExpiry` at line 240? Why do we care if the nil session has expired? The semantics of the nil session are that there is no session.

Reply:
> Why do we care if the nil session has expired? The semantics of the nil session are that there is no session.
I had considered a nil session to still be a valid session. Refactored as this was understanding was incorrect. 

> Isn't this always true, since we've updated this.sessionExpiry at line 240?
Removed check as this is now obsolete.